transactions/WFLY-14869-microprofile-lra.adoc

```diff
@@ -0,0 +1,106 @@
+= Support for MicroProfile LRA
+:author:            Ondrej Chaloupka
+:email:             ochaloup@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+:keywords:          transaction, saga, lra, microprofile
+
+== Overview
+
+The goal of this RFE is providing the Galleon feature pack for WildFly
+which integrates the https://github.com/jbosstm/narayana/tree/5.11.2.Final/rts/lra[Narayana implementation]
+of the https://github.com/eclipse/microprofile-lra[MicroProfile LRA specification].
+
+The MicroProfile LRA is a transaction processing pattern well suited for microservice development
+based on the saga pattern. The motivation of such processing is summarized in the https://github.com/eclipse/microprofile-lra/blob/1.0/README.adoc#motivation[MicroProfile Spec readme].
+
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-14869
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1496
+
+=== Dev Contacts
+
+* mailto:{mkunwar@redhat.com}[Mayank Kunwar]
+* mailto:{mstefank@redhat.com}[Martin Stefanko]
+* mailto:{ochaloup@redhat.com}[{author}]
+
+=== QE Contacts
+
+* TBD
+
+=== Testing By
+
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+* [ ] QE
+
+TBD
+
+=== Affected Projects or Components
+
+* Transactions
+* WildFly
+
+=== Other Interested Projects
+
+* MicroProfile
+* Galleon
+
+=== Relevant Installation Types
+
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+* [ ] Managed domain
+* [ ] OpenShift s2i
+* [ ] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+The feature pack will integrate the LRA specification as the Galleon feature pack.
+The Galleon layer will be available for provisioning.
+
+Each MP spec is optional to allow for exclusion during provisioning.
+
+The provisioned feature pack will provide all functionality defined in LRA specification for users in WildFly.
+
+The LRA feature pack will be dependent on the `transactions` subsystem. The LRA uses the same object store as the JTA/JTS transaction uses,
+it requires the recovery manager thread to be running and requires transaction specifics like unique node-identifier per running app server.
+
+=== Nice-to-Have Requirements
+
+NONE
+
+=== Non-Requirements
+
+NONE
+
+== Test Plan
+
+1. Testing LRA functionality within WildFly
++
+The LRA functionality tests are part of the Narayana LRA codebase. The WildFly will be running with LRA TCK module (https://github.com/jbosstm/narayana/tree/5.11.2.Final/rts/lra/test/tck)
```

Comment:
[Some work](https://issues.redhat.com/browse/JBTM-3427) towards this has been already carried out but it is not completed yet

Reply:
+1